_search-plugins/knn/knn-vector-quantization.md

```diff
@@ -151,9 +181,72 @@ As an example, assume that you have 1 million vectors with a dimension of 256 an
+quantization works by breaking up vectors into _m_ subvectors, and  encoding each subvector with _code_size_ bits. Thus,
+the total amount of memory for the vector ends up being `m*code_size` bits, plus overhead. For more details about the 
+parameters of product quantization, see 
+[PQ parameters]({{site.url}}{{site.baseurl}}/search-plugins/knn/knn-index/#pq-parameters). Product quantization is only 
+supported for the _Faiss_ engine and can be used with either the _HNSW_ or the _IVF_ ANN algorithms.
+
+### Using Faiss product quantization
+
+In order to minimize the loss in accuracy, product quantization requires a _training_ step that builds a model based on 
+the distribution of the data that will be searched over.
+
+Under the hood, the product quantizer is trained by running k-Means clustering on a set of training vectors for each 
+sub-vector space and extracts the centroids to be used for the encoding. The training vectors can either be a subset 
+of the vectors to be ingested, or vectors that have the same distribution and dimension as the vectors to be ingested. 
+In OpenSearch, the training vectors need to be present in an index. In general, the amount of training data will depend 
+on which ANN algorithm will be used and how much data will go into the index. For IVF-based indices, a good number of 
+training vectors to use is `max(1000*nlist, 2^code_size * 1000)`. For HNSW-based indices, a good number is 
+`2^code_size*1000` training vectors. See [Faiss's documentation](https://github.com/facebookresearch/faiss/wiki/FAQ#how-many-training-points-do-i-need-for-k-means) 
+for more details on how these numbers are arrived at.
+
+For product quantization, the two parameters that need to be selected are _m_ and _code_size_. _m_ determines how many 
+sub-vectors the vectors should be broken up into to encode separately - consequently, the _dimension_ needs to be 
+divisible by _m_. _code_size_ determines how many bits each sub-vector will be encoded with. In general, a good place to
+start is setting `code_size = 8` and then tuning _m_ to get the desired tradeoff between memory footprint and recall.
+
+For an example of setting up an index with product quantization, see [this tutorial]({{site.url}}{{site.baseurl}}/search-plugins/knn/approximate-knn/#building-a-k-nn-index-from-a-model).
+
+### Memory Estimation
+
+While product quantization is meant to represent individual vectors with `m*code_size` bits, in reality the indices 
+take up more space than this. This is mainly due to the overhead of storing certain code tables and auxilary data 
+structures.
+
+Some of the memory formulas depend on the number of segments present. Typically, this is not known beforehand but a good
+default value is 300.
+{: .note}
+
+#### HNSW memory estimation
+
+The memory required for HNSW with PQ is estimated to be `1.1*((pq_code_size / 8 * pq_m + 24 + 8 * hnsw_m) * num_vectors + num_segments * (2^pq_code_size * 4 * d))` bytes.
```

Comment:
Should we add parenthesis for each component to avoid operator precedence issues for evaluating memory? For example (pq_code_size/8)*pq_m

Reply:
I think it makes sense with respect to the "/" operation. For others, it may lead to too many parenthesis. I updated